theseus/geometry/so3.py

```diff
@@ -166,12 +176,14 @@ def _log_map_impl(
         return SO3_base.log(self.tensor, jacobians=jacobians)
 
     def _compose_impl(self, so3_2: LieGroup) -> "SO3":
-        return SO3(tensor=SO3_base.compose(self.tensor, so3_2.tensor))
+        return SO3(
+            tensor=SO3_base.compose(self.tensor, so3_2.tensor), disable_checks=True
```

Comment:
Wondering if we should disable checks for `compose()` since composition is one of the major sources for the accumulation of numerical errors.